src/components/GameDetail/GameProgress.vue

```diff
@@ -0,0 +1,279 @@
+<template lang="html">
+  <div :class="['game-progresses', { small }]">
+    <div v-if="hasProgress && !showProgressField" class="progress">
+      <div class="progress-data">
+        <progress
+          class="progress-bar"
+          max="100"
+          :value="localProgress.number"
+        >
+          <div class="progress-bar-fallback">
+            <span :style="style">
+              Progress: {{localProgress.number}}%
+            </span>
+          </div>
+        </progress>
+        <div class="progress-bar-label">
+          {{localProgress.number}}%
+        </div>
+      </div>
+
+      <button
+        v-if="!small"
+        class="primary"
+        @click="editProgress"
+      >
+        Edit progress
+      </button>
+    </div>
+
+    <div
+      class="progress-field"
+      v-if="showProgressField && !small"
+    >
+      <input
+        v-model.number="localProgress.number"
+        placeholder="50"
+        type="number"
+        autofocus
+      >
```

Comment:
consider using [range](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range) for the edit view, could be more intuitive for touch users